homeassistant/components/touchline/climate.py

```diff
@@ -95,8 +124,91 @@ def target_temperature(self):
         """Return the temperature we try to reach."""
         return self._target_temperature
 
+    @property
+    def current_operation_mode(self):
+        """Return the current operation mode."""
+        return self._current_operation_mode
+
+    @property
+    def current_week_program(self):
+        """Return the current week program."""
+        return self._current_week_program
+
+    @property
+    def current_preset_mode(self):
+        """Return the current preset mode."""
+        return self._current_preset_mode
+
+    @property
+    def preset_mode(self):
+        """Return the current preset mode."""
+        return self._preset_mode
+
+    @property
+    def preset_modes(self):
+        """Return available preset modes."""
+        return PRESET_MODES
+
+    def set_preset_mode(self, preset_mode):
+        """Set new target preset mode."""
+        mode = self.map_mode_hass_touchline(preset_mode)
+        program = self.map_program_hass_touchline(preset_mode)
+        self.unit.set_operation_mode(mode)
+        self.unit.set_week_program(program)
+        self._current_operation_mode = mode
+        self._current_week_program = program
+        self._current_preset_mode = preset_mode
+
     def set_temperature(self, **kwargs):
         """Set new target temperature."""
         if kwargs.get(ATTR_TEMPERATURE) is not None:
             self._target_temperature = kwargs.get(ATTR_TEMPERATURE)
         self.unit.set_target_temperature(self._target_temperature)
+
+    @staticmethod
+    def map_mode_hass_touchline(preset_mode):
```

Comment:
Please use a module level constant dictionary instead to map between the device and home assistant modes.